### DISCUSSION TITLE
Reports giving NaN result while integrating into Cypress typescript project

### DISCUSSION THREAD

**Discussion by mariamaslam** [mmisty/cypress-allure-adapter#78] [Q&A] [answered]
I am trying to use this in cypress typescript, this worked prefectly fine on cypress javacript.

```
import { defineConfig } from "cypress";

import { configureAllureAdapterPlugins } from '@mmisty/cypress-allure-adapter/plugins';


export default defineConfig({
  e2e: {
    baseUrl: "https://locahost.com/",
    supportFile: "cypress/support/commands.ts",

    env: {
      allure: true,
      allureCleanResults: true,
        allureSkipCommands: 'wrap',
        allureResults: '../../allure-results',
    },
    setupNodeEvents(on, config) {
          configureAllureAdapterPlugins(on, config);
          const reporter = configureAllureAdapterPlugins(on, config);
      
          console.log(' === ENVIRONMENT:');
          console.log(config.env);
          console.log(' === ');
      
          // this is to write categories and environment information
          on('before:run', details => {
            reporter?.writeEnvironmentInfo({
               info: {
                  os: details.system.osName,
                  osVersion: details.system.osVersion,
              },
            });
      
             reporter?.writeCategoriesDefinitions({ categories: './allure-error-categories.json' });
          });
          
          
          // important to return config
          return config;
        },
        }
      })
```

```
import './commands'
import '@mmisty/cypress-allure-adapter/support';
```

```

"scripts": {
    "test": "cypress run cypress/e2e/Registration/*.cy.ts && npm run html report",
    "cy:run": "cypress run cypress/e2e/*.cy.ts --env allure=true || true && allure generate allure-results --clean -o allure-report && allure open allure-report",
    "allure:report": "npm run allure generate allure-result && allure open allure-result",
    "beforetest": "npm run clean:folders",
    "tests": "npm run cy:run || npm run aftertest",
    "aftertest": "npm run allure:report"
  },
```
      

The results are NaN

**Comment by mariamaslam** [comment]
My allure-result folder is empty

**Comment by mmisty** [comment]
hi @mariamaslam 

You have allureResults set to ../../allure-results. Is it done by intention?

could you do steps that I've recommended in cypress thread. 


1. please delete allure-results folder
2. run tests by cypress run --env allure=true and verify they were executed
3. check allure-results folder: do you have *.json files in your allure-results folder?

If you do have *.json files in your allure-results folder after tests executed that means that plugin works correctly and most likely you are generating report incorrectly.

Otherwise, if allure-results folder is created but empty I will need some time (since I'm away from computer at the moment)

**Comment by mariamaslam** [comment]
Hello,
thank you for the prompt response @mmisty 
I did execute the steps you mentioned; it doesn't seem to be working. I've attached the screenshot of my allure-result.
No worries, I look forward to your response when you are available

<img width="469" alt="Screenshot 2023-11-11 at 3 32 40 PM" src="https://github.com/mmisty/cypress-allure-adapter/assets/47422016/3038c0fc-ed0d-4d43-bdc7-d723efe82dc7">


<img width="610" alt="Screenshot 2023-11-11 at 3 37 03 PM" src="https://github.com/mmisty/cypress-allure-adapter/assets/47422016/758870e4-7634-4be2-9341-951dc74f066c">

**Reply by mmisty** [reply]
@mariamaslam  Once again -  you need to check where you do write `allure-results` - and it seems you `allure-results` folder is outside of your project according to your configuration `allureResults: '../../allure-results'` and you are building report from directory within your project - which of course will be NaN since no allure results in the folder.

**Comment by mariamaslam** [comment]
hi @mmisty i look forward to your response. please let me know possibility of how soon can youget back to me?

**Comment by mmisty** [comment, marked as answer]
Hi @mariamaslam 

I created working [ ts example](https://github.com/mmisty/cypress-allure-adapter-example/tree/main/examples/cypress-ts) 
Report: 
<img width="1197" alt="image" src="https://github.com/mmisty/cypress-allure-adapter/assets/16957275/02e3844c-b715-42ef-aaca-ac82ea0d93df">

Proposing to compare the configuration with yours.
If some issues  - checkout repo and try to play with it

**Comment by mariamaslam** [comment]
Hi am grateful, i will try this shortly n update you.

**Reply by mariamaslam** [reply]
I tried my level best but im still getting this error
`[cypress-allure-adapter] Categories file doesn't exist './allure-error-categories.json'`

The allure-result folder is empty.

**Reply by mmisty** [reply]
@mariamaslam this error has nothing with your problem (but if you don't want to see it delete line .`reporter?.writeCategoriesDefinitions({ categories: './allure-error-categories.json' });` from cypress.config.ts) 

Could you verify that you've checked what I suggested above in the thread: 
> @mariamaslam Once again - you need to check where you do write allure-results - and it seems you allure-results folder is outside of your project according to your configuration allureResults: '../../allure-results' and you are building report from directory within your project - which of course will be NaN since no allure results in the folder.

So, you need to fix your configuration from :
`allureResults: '../../allure-results',`
to 
`allureResults: 'allure-results',`

Also, I proposing to use **ChatGPT**.  It can help a lot.

**Reply by mmisty** [reply]
@mariamaslam JFYI, running tests by the command `cypress run cypress/e2e/*.cy.ts` will not select tests by pattern `cypress/e2e/*.cy.ts`  - it will run all tests. To run tests by pattern you need to fix the script
```
cypress run --config specPattern="cypress/e2e/*.cy.ts"
```

**Reply by mariamaslam** [reply]
Could it be possible that this is the error that is causing the issue?

` "[allure-js-parser] No allure-results in folder (did you forgot to run tests?). Path 'allure-results'"
`
yup, I did try chatgpt but since your plugin is new it keeps saying to install shelex one.

Moreover, I did follow the steps above but no luck, the allure-result folder is empty

package.json

```
"scripts": {
    "test": "cypress run cypress/e2e/Registration/*.cy.ts --env allure=true",
    "cy:run": "cypress run--config specPattern=cypress/e2e/*.cy.ts --env allure=true || true && allure generate allure-results --clean -o allure-report && allure open allure-report",
    "allure:report": "npm run allure generate allure-results && allure open allure-results",
    "beforetest": "npm run clean:folders",
    "tests": "npm run cy:run || npm run aftertest",
    "aftertest": "npm run allure:report"
  },
```


cypress.config.ts
```

import { defineConfig } from "cypress";
import path from "path";
import gmail from "gmail-tester-extended";
import { configureAllureAdapterPlugins } from '@mmisty/cypress-allure-adapter/plugins';


module.exports = defineConfig({
  e2e: {
    defaultCommandTimeout: 1500,
    video: true,
    baseUrl: "https://dev.local.com/",
    supportFile: "cypress/support/commands.ts",

    env: {
      allure: true,
      allureCleanResults: false,
      allureSkipCommands: 'wrap',
      allureResults: 'allure-results',
    },
    specPattern: [
      'cypress/e2e/Registration/Signup.cy.ts',
      'cypress/e2e/Registration/EmailVerification.cy.ts',
      'cypress/e2e/Registration/ResendEmail.cy.ts',
      'cypress/e2e/Registration/Login.cy.ts',
    ],
    setupNodeEvents(on, config) {
          on("task", {
            "gmail:get-all-emails": async args => {
              const { subject, from, to, snippet, message} = args;
              const allEmails = await gmail.get_messages(
                path.resolve(__dirname, "OAuth.json"),
                path.resolve(__dirname, "AccessToken.json"),
                {
                  subject: subject,
                  from: from,
                  to: to,
                  include_body: true,
                  snippet: snippet,
                  message: message,
                }
              );
              return allEmails;
            }
          });
          
          const reporter = configureAllureAdapterPlugins(on, config);

          console.log(' === ENVIRONMENT:');
          console.log(config.env);
          console.log(' === ');
      
          // this is to write categories and environment information
          on('before:run', details => {
            reporter?.writeEnvironmentInfo({
               info: {
                  os: details.system.osName,
                  osVersion: details.system.osVersion,
              },
            });
      
             reporter?.writeCategoriesDefinitions({ categories: './allure-error-categories.json' });
          });
          
          
          // important to return config
          return config;
        },
        }
      })

```

      
Tha allure-results folder is empty and there is no json file



<img width="335" alt="Screenshot 2023-11-14 at 6 48 38 AM" src="https://github.com/mmisty/cypress-allure-adapter/assets/47422016/a7347328-61ed-4181-8722-7656a94ba403">

**Comment by mariamaslam** [comment]
Im just so surprised that same solution is working on Cypress JS but not Cypress TS. Everything is same there is no difference?
Do i need to define anything in config.json  in TS?

**Reply by mmisty** [reply]
@mariamaslam  I see that you have this is your config
```
supportFile: "cypress/support/commands.ts",
```
Are you sure that this is right?   Since I see other file(which I don't know how named) with 
```
import './commands'
import '@mmisty/cypress-allure-adapter/support';
```
Please set support file to a correct one (that has `import '@mmisty/cypress-allure-adapter/support';` inside)

 If that still doesn't work  please send the log of running tests by 
`DEBUG=cypress-allure* cypress run`
